api/provider/aws/environment_delete.go

```diff
@@ -51,32 +52,33 @@ func (e *EnvironmentProvider) Delete(environmentID string) error {
 		}
 
 		// Check for eventually consistency
-		fn := func() (shouldRetry bool, err error) {
+		var err error
```

Comment:
You have already done this logic above with:
```go
if securityGroup != nil {
    groupID := aws.StringValue(securityGroup.GroupId)
    if err := deleteSGWithRetry(e.AWS.EC2, groupID); err != nil {
        return err
    }
...
```

That whole block can end after the `deleteSGWithRetry(e.AWS.EC2, groupID)` call; this extra retry stuff is just redundant.